server/zanata-war/src/test/java/org/zanata/tmx/TMXParserTest.java

```diff
@@ -208,6 +212,14 @@ public void parseTMX() throws Exception {
         for (TransMemoryUnit tu : tm.getTranslationUnits()) {
             assertThat(tu.getTransUnitVariants().size(), greaterThan(0));
         }
+
+        Optional<String> seg = tm.getTranslationUnits().stream()
+                .filter(tu -> tu.getTransUnitId() != null && tu.getTransUnitId().equals("tuid2"))
+                .map(TransMemoryUnit::getTransUnitVariants)
+                .map(map ->  map.get("ja"))
+                .map(TransMemoryUnitVariant::getPlainTextSegment)
+                .findAny();
+        assertThat(seg.orElseGet(null), containsString("を選択します"));
```

Comment:
Not sure we need this assertion here. It was not detecting the issue last I remembered.